services/frontend/domain/usecases/process_manifest_files.usecase.go

```diff
@@ -0,0 +1,154 @@
+package usecases
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v3"
+
+	"github.com/berops/claudie/internal/manifest"
+	"github.com/berops/claudie/proto/pb"
+)
+
+// ProcessManifestFiles processes the manifest files concurrently. If an error occurs while the file
+// is being processed, it's skipped and the function continues with the next one until all files are
+// processed. Nothing is done with those files for which errors occurred, they'll be skipped until either
+// corrected or deleted.
+func (u *Usecases) ProcessManifestFiles(manifestDir string) error {
+
+	manifestFiles, err := os.ReadDir(manifestDir)
+	if err != nil {
+		return fmt.Errorf("Failed to read manifest files from dir %q: %w", manifestDir, err)
+	}
+
+	configs, err := u.ContextBox.GetAllConfigs()
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve manifests details from context-box: %w", err)
+	}
+
+	log.Debug().Msgf("%d configs present in database | %d configs in %v", len(configs), len(manifestFiles), manifestDir)
+
+	manifestProcessingResultsChan := make(chan *ManifestProcessingResult, len(manifestFiles))
+
+	waitGroup := sync.WaitGroup{}
+
+	for _, manifestFile := range manifestFiles {
+		waitGroup.Add(1)
+
+		// Process each of the files concurrently in a separate go-routine skipping over files for which
+		// an error occurs.
+		// By processing, we mean unmarshalling the claudie manifest
+		go func(manifestFile fs.DirEntry) {
+			defer waitGroup.Done()
+
+			manifestFilepath := filepath.Join(manifestDir, manifestFile.Name())
+
+			var (
+				rawManifestData []byte
+				manifest        manifest.Manifest
+				processingError error
+			)
+
+			defer func() {
+				manifestProcessingResultsChan <- &ManifestProcessingResult{
+					manifestName:     manifest.Name,
+					rawManifestData:  rawManifestData,
+					manifestFilepath: manifestFilepath,
+					processingError:  processingError,
+				}
+			}()
+
+			if rawManifestData, processingError = os.ReadFile(manifestFilepath); err != nil {
+				return
+			}
+
+			if processingError = yaml.Unmarshal(rawManifestData, &manifest); processingError != nil {
+				return
+			}
+
+			manifest.Validate()
+		}(manifestFile)
+	}
+
+	go func() {
+		waitGroup.Wait()
+
+		close(manifestProcessingResultsChan)
+	}()
+
+	// Collect processing results of manifest files which were processed successfully
+	for manifestProcessingResult := range manifestProcessingResultsChan {
+		// TODO: review
+		if manifestProcessingResult.processingError != nil {
+			log.Error().Msgf("Skipping over file %v due to processing error : %v", manifestProcessingResult.manifestFilepath, manifestProcessingResult.processingError)
+			continue
+		}
+
+		configs = removeConfig(configs, manifestProcessingResult.manifestName)
```

Comment:
This function has to be called first in the loop, as any configs left over will be deleted.

If we leave this line here, after checking for errors in processing, the configs which had an error will get deleted, which we do not want.